lib/server.js

```diff
@@ -386,6 +387,91 @@ module.exports = function createServer(config, log) {
     }
   )
 
+  api.post('/checkCustomRule', function(req, res, next) {
+    const action = req.body.action
+    const ip = req.body.ip
+    const email = req.body.email
+
+    if (!ip || !action || !email) {
+      const err = {code: 'MissingParameters', message: 'action, ip email, are all required'}
+      log.error({op: 'request.checkCustomRule', action, ip, email})
+      res.send(400, err)
+      return next()
+    }
+
+    // Get all the custom rules that might apply to this action
+    const customRules = config.customRules
+    const checkRules = Object.keys(customRules).filter((key) => {
+      return customRules[key].actions.indexOf(action) > -1
+    })
+    if (checkRules.length <= 0) {
+      const err = {code: 'MissingParameters', message: 'no rules found for ' + action}
+      log.error({op: 'request.checkCustomRule', action, ip, email})
+      res.send(400, err)
+      return next()
+    }
+
+    // Retrieve the record for each rule and check if `action` has been rate limited
+    const recordResults = []
+    checkRules.forEach((ruleName) => {
+      const recordKey = ip + email + ruleName
```

Comment:
If it's a relatively small change, let's consider hashing this down to something with a known fixed length, to avoid https://bugzilla.mozilla.org/show_bug.cgi?id=1365486